react-app/src/components/ConfigForm.js

```diff
@@ -1,7 +1,59 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Alert, Button } from 'react-bootstrap';
 
-function ConfigForm() {
-  return <p className="page-text text-centered">The config editor form will display here.</p>;
+function ConfigForm(props) {
+  const [showSavedAlert, setShowSavedAlert] = useState(false);
+  const [showErrorAlert, setShowErrorAlert] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  function onSaveAs() {
+    window.api
+      .saveConfigAs(props.schema)
+      .then((result) => {
+        if (result === true) {
+          // if saveOutput() returns true, then the save process succeeded
+          setShowSavedAlert(true);
+        } else if (typeof result === 'string') {
+          // if the result is a string, that means the save process returned an error message
+          setErrorMessage(result);
+          setShowErrorAlert(true);
+        }
```

Comment:
Is there a reason the saveConfigAs function doesn't just always throw an error when one occurs? do we have control over that api? Could we just be throwing an error if an error occurs and catching it in the catch block below?

Reply:
I've fixed this.